source/non_matching/coupling.cc

```diff
@@ -97,6 +97,9 @@ namespace NonMatching
               &immersed_dh.get_triangulation()) == nullptr),
            ExcNotImplemented());
 
+    bool optimize_mpi =
+      (dynamic_cast<const parallel::Triangulation<dim1, spacedim> *>(
+         &space_dh.get_triangulation()) != nullptr);
```

Comment:
I think it would help if we used a name for the boolean that reflects this check, e.g., `tria_is_parallel`.

Reply:
You should be able to make this variable `const`